graphics/image-archive.h

```diff
@@ -0,0 +1,56 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef GRAPHICS_IMAGE_ARCHIVE_H
+#define GRAPHICS_IMAGE_ARCHIVE_H
+
+#include "common/hashmap.h"
+#include "common/path.h"
+
+namespace Common {
+class Archive;
+}
+
+namespace Graphics {
+struct Surface;
+
+class ImageArchive {
+public:
+	~ImageArchive();
+
+	void reset();
```

Comment:
Since now it is in the common code, it would be good to add Doxygen documentation here

Reply:
I've added some brief comments for the class and the public functions.